src/app/modules/map/components/map/map.component.ts

```diff
@@ -540,81 +524,36 @@ export class MapComponent implements OnInit, OnDestroy, AfterViewInit {
       : settings.map.tiles.maxZoom;
   }
 
-  private getTileLayerFactory(topoMap: TopoMap): CreateTileLayer[] {
-    let createNorwegianMixedMap: CreateTileLayer;
-    let createStatensKartverk: CreateTileLayer;
-
-    if (isAndroidOrIos(this.platform)) {
-      createNorwegianMixedMap = (options) => new RegObsOfflineAwareTileLayer(
-        TopoMap.statensKartverk,
-        settings.map.tiles.statensKartverkMapUrl,
-        {
-          ...options,
-          bounds: settings.map.tiles.supportTilesBounds as L.LatLngBoundsLiteral
-        },
-        this.offlineMapService.offlineTilesRegistry,
-        this.loggingService,
-      );
-
-      createStatensKartverk = (options) => new RegObsOfflineAwareTileLayer(
-        TopoMap.statensKartverk,
-        settings.map.tiles.statensKartverkMapUrl,
-        options,
-        this.offlineMapService.offlineTilesRegistry,
-        this.loggingService,
-      );
-    } else {
-      createNorwegianMixedMap = (options) => new RegObsTileLayer(
-        settings.map.tiles.statensKartverkMapUrl,
-        {
-          ...options,
-          bounds: settings.map.tiles.supportTilesBounds as L.LatLngBoundsLiteral
-        }
-      );
-
-      createStatensKartverk = (options) => new RegObsTileLayer(
-        settings.map.tiles.statensKartverkMapUrl,
-        options
-      );
-    }
-
-
-    const createOpenTopoMap: CreateTileLayer = (options) => new L.TileLayer(settings.map.tiles.openTopoMapUrl, options);
-    const createArcGisOnlineMap: CreateTileLayer = (options) => new L.TileLayer(settings.map.tiles.arcGisOnlineTopoMapUrl);
-    const createGeoDataLandskapMap: CreateTileLayer = (options) => new L.TileLayer(settings.map.tiles.geoDataLandskapMapUrl);
-    const createArGisOnlineMixMap: CreateTileLayer[] = [
-      (options) => new RegObsTileLayer(
-        settings.map.tiles.arcGisOnlineTopoMapUrl,
-        {
-          ...options,
-          zIndex: MapLayerZIndex.OnlineMixedBackgroundLayer,
-          excludeBounds: NORWEGIAN_BOUNDS
-        },
-      ),
-      createNorwegianMixedMap
-    ];
-
-    switch (topoMap) {
-    case TopoMap.statensKartverk:
-      return [createStatensKartverk];
-    case TopoMap.openTopo:
-      return [createOpenTopoMap];
-    case TopoMap.arcGisOnline:
-      return [createArcGisOnlineMap];
-    case TopoMap.geoDataLandskap:
-      return [createGeoDataLandskapMap];
-    case TopoMap.mixOpenTopo:
-      return [
-        (options) => new RegObsTileLayer(settings.map.tiles.openTopoMapUrl, {
-          ...options,
-          zIndex: MapLayerZIndex.OnlineMixedBackgroundLayer,
-          excludeBounds: NORWEGIAN_BOUNDS
-        }),
-        createNorwegianMixedMap
-      ];
-    case TopoMap.mixArcGisOnline:
-    default:
-      return createArGisOnlineMixMap;
+  private *getBaseMapLayers(topoMap: TopoMap, useRetinaMap: boolean) {
```

Comment:
Jeg synes denne måten å hente hvilke lag som skal vises var mye lettere å forstå enn slik det var før. 
Har bare en par spørsmål:
- Denne * foran funksjonsnavnet var ny for meg. Har den noe med yield å gjøre lengre ned?
- Hva er fordelen med å bruke yield framfor å legge til en liste og returnere denne?
- Bør du deklarere hvilken type som returneres?

Reply:
Det er bare en fancy måte å slippe å lage en liste på og returnere denne til slutt. Samme for meg hvordan vi gjør det, nytten av en generator her er svært begrensa, kun litt mindre kode, kanskje mindre lesbart. En generator er best til pågående beregninger der man vil slippe å bruke mye minne på å håndtere resultater underveis og gjerne vil ha resultatene fortløpende. Du kan lese mer om generators her: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Generator . En generator kan itereres over, så i `configureTileLayers` kan vi bare bruke den på samme måte som andre ting man kan iterere over, feks arrays.